app/controllers/discovered_hosts_controller.rb

```diff
@@ -196,7 +196,11 @@ def init_regex_and_categories
     @categories_names = [:Hightlights, :Storage, :Hardware, :Network, :Software, :IPMI, :Misceleneous]
   end
 
-  def assign_fact_to_category(key, value )
+  def assign_fact_to_category(key, value)
+    if @interfaces.any? {|interface| key.include? interface[:identifier]}
```

Comment:
That's right, this should be in a class, with unit testing. Created http://projects.theforeman.org/issues/14229

For now, I have added functional test that at least excercise the page if it renders properly. We haven't had one. Rebased.